### PR TITLE
Add record limitation

### PR DESCRIPTION
**Pull request by shyeavr**
add get_folder_last_modified_content for the swap update test validation process

### CODE REVIEW

File: mc_automation_tools/base_requests.py [hunk 1]
```diff
@@ -50,9 +55,11 @@ def send_post_binary_request(
...
 
 
-def send_post_request(url, body={}, header=None, params=None):
+def send_post_request(url, body=None, header=None, params=None):
     """send http post request by providing post full url + body , header is optional, by default:content-type': 'application/json',
     "accept": "*/*"""
+    if body is None:
+        body = {}
     common.url_validator(url)
     if not header:
```
Comment (dannysigalovich) [inline mc_automation_tools/base_requests.py:64]: if headers is None ... then do it

File: mc_automation_tools/base_requests.py [hunk 2]
```diff
@@ -50,9 +55,11 @@ def send_post_binary_request(
     return resp
 
 
-def send_post_request(url, body={}, header=None, params=None):
+def send_post_request(url, body=None, header=None, params=None):
```
Comment (dannysigalovich) [inline mc_automation_tools/base_requests.py:58]: Where is the requests.post ? you are not sending post request i dont see it

File: setup.py [hunk 3]
```diff
@@ -1,3 +1,30 @@
+# import setuptools
```
Comment (dannysigalovich) [inline setup.py]: remove dead code.

File: mc_automation_tools/validators/mapproxy_validator.py [hunk 4]
```diff
@@ -57,7 +54,7 @@ def validate_layer_from_pycsw(
         links = self.extract_from_pycsw(pycsw_records)
```
Comment (dannysigalovich) [inline mc_automation_tools/validators/mapproxy_validator.py]: you can use "items" maybe better 
maybe like this 
 for product_type, product_links in links.items():
        for li, link_value in product_links.items():
           if li in ["WMS", "WMTS_KVP"]:
                links[product_type][li] += f"&token={token}"
            elif li == "WMTS_BASE":
                links[product_type][li] += f"/{product_id}-{product_type}"

File: mc_automation_tools/base_requests.py [hunk 5]
```diff
@@ -156,11 +163,28 @@ def send_delete_request(url, params=None):
     try:
```
Comment (dannysigalovich) [inline mc_automation_tools/base_requests.py:163]: Where is the exception ? Except requests.RequestException as e: 
like this ? 
 except requests.RequestException as e:
        _log.error("Request failed: %s", str(e))
        raise

**Review by dannysigalovich — CHANGES_REQUESTED** [review]

File: mc_automation_tools/base_requests.py (hunk 2 above)
Comment (shyeavr) [reply mc_automation_tools/base_requests.py:58]: solve

**Review by shyeavr — COMMENTED** [review]
done

**Pull request merged**